srp/src/utils.rs

```diff
@@ -38,6 +55,33 @@ pub fn compute_m1<D: Digest>(a_pub: &[u8], b_pub: &[u8], key: &[u8]) -> Output<D
     d.finalize()
 }
 
+#[must_use]
+pub fn compute_hash<D: Digest>(data: &[u8]) -> Output<D> {
+    let mut d = D::new();
+    d.update(data);
+    d.finalize()
+}
+
+// M1 = H(H(N) XOR H(g) | H(U) | s | A | B | K) this follows the spec
+#[must_use]
+pub fn compute_m1_std<D: Digest>(
```

Comment:
"Spec" doesn't say which spec, and `std` in particular is easily confused with the Rust standard library.

Perhaps use  RFC5054 instead?

```suggestion
/// M1 = H(H(N) XOR H(g) | H(U) | s | A | B | K) following RFC5054
#[must_use]
pub fn compute_m1_rfc5054<D: Digest>(
```